src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/MsQuicException.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Net.Quic;
+using System.Net.Sockets;
+using static Microsoft.Quic.MsQuic;
+
+namespace Microsoft.Quic
+{
+    internal class MsQuicException : QuicException
```

Comment:
I'm not 100% convinced about this, we could merge this with `QuicException`. I just wanted to keep msquic return error based exception separate. Or can deal with this when we discus more broader exception topics like #55619 and #56954